pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -709,15 +712,15 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 			ps.CheckMaxTimeTopic(agentName, "routeChanges", start,
 				warningTime, errorTime)
 
-		case <-publishTimer.C:
+		case <-zedrouterCtx.publishTicker.C:
 			start := time.Now()
-			log.Traceln("publishTimer at", time.Now())
-			err := pub.Publish("global",
-				getNetworkMetrics(&zedrouterCtx))
+			log.Traceln("publishTicker at", time.Now())
+			nms := getNetworkMetrics(&zedrouterCtx)
+			err := pub.Publish("global", nms)
 			if err != nil {
 				log.Errorf("getNetworkMetrics failed %s\n", err)
 			}
-			publishNetworkInstanceMetricsAll(&zedrouterCtx)
+			publishNetworkInstanceMetricsAll(&zedrouterCtx, &nms)
```

Comment:
Why do you need to/want to reuse nms here?
I think the functions add data to the nms struct so you report/count things twice??

But my main question is what is broken in the current code taht requires a fix?

Reply:
We call getNetworkMetrics() twice (here and inside of publishNetworkInstanceMetricsAll) which is sub-optimal as we call iptables binary internally several times on every call.

Reply:
OK I see that now, but it is a bit brittle to assume that the caller passes in the unmodified value from publishNetworkInstanceMetricsAll. In the future some code might be added to augment the metrics before the publish.
Can you add some comments to the caller saying that the nms must be exactly the unmodified output from getNetworkMetrics()?